### PR TITLE
Private link endpoint sql 101

### PR DESCRIPTION
**Pull request by mblanco77**
# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-privatelink-endpoint-sql/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        "osProfile": {
+          "computerName": "[variables('vmName')]",
+          "adminUsername": "[parameters('adminUsername')]",
+          "adminPassword": "[parameters('adminPassword')]"
+        },
+        "storageProfile": {
+          "imageReference": {
+            "publisher": "MicrosoftWindowsServer",
+            "offer": "WindowsServer",
+            "sku": "2016-Datacenter",
```
Comment (bmoore-msft) [inline 101-privatelink-endpoint-sql/azuredeploy.json:270]: any reason we can't use 2019?

File: 101-privatelink-endpoint-sql/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,295 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "administratorLogin": {
```
Comment (bmoore-msft) [inline 101-privatelink-endpoint-sql/azuredeploy.json:5]: change the parameter names for the vm & sql so it's clear by the name which params are used for which resources

File: 101-privatelink-endpoint-sql/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+  },
+  "variables": {
+    "vnetName": "myVirtualNetwork",
+    "vnetAddressPrefix": "10.0.0.0/16",
+    "subnet1Prefix": "10.0.0.0/24",
+    "subnet1Name": "mySubnet",
+    "sqlServerName": "[concat('sqlserver', uniqueString(resourceGroup().id))]",
+    "databaseName": "[concat(variables('sqlServerName'),'/sample-db')]",
+    "privateEndpointName": "myPrivateEndpoint",
+    "privateDnsZoneName": "privatelink.database.windows.net",
```
Comment (bmoore-msft) [inline 101-privatelink-endpoint-sql/azuredeploy.json:52]: this looks to be specific for public Azure - if it's not, then use something generic so it doesn't look "wrong" for Azure Government

File: 101-privatelink-endpoint-sql/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+          ],
+          "properties": {
+            "addressPrefix": "[variables('subnet1Prefix')]",
+            "privateEndpointNetworkPolicies": "Disabled"
+          }
+        }
+      ]
+    },
+    {
+      "location": "[parameters('location')]",
```
Comment (bmoore-msft) [inline 101-privatelink-endpoint-sql/azuredeploy.json:132]: see the BP for ordering property 
https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties

File: 101-privatelink-endpoint-sql/azuredeploy.json (hunk 3 above)
Comment (mblanco77) [reply 101-privatelink-endpoint-sql/azuredeploy.json:52]: Hi this is the Private DNS Zone name that is suggested for the Azure SQL private endpoint
https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-dns#azure-services-dns-zone-configuration

File: 101-privatelink-endpoint-sql/azuredeploy.json (hunk 1 above)
Comment (mblanco77) [reply 101-privatelink-endpoint-sql/azuredeploy.json]: done in 05b1977

File: 101-privatelink-endpoint-sql/azuredeploy.json (hunk 2 above)
Comment (mblanco77) [reply 101-privatelink-endpoint-sql/azuredeploy.json]: done in 05b1977

File: 101-privatelink-endpoint-sql/azuredeploy.json (hunk 4 above)
Comment (mblanco77) [reply 101-privatelink-endpoint-sql/azuredeploy.json]: done in 05b1977

File: 101-privatelink-endpoint-sql/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [reply 101-privatelink-endpoint-sql/azuredeploy.json:52]: I'm not sure if that means the doc should be updated or we expect sovereign clouds top point to public Azre?

Comment (mblanco77) [reply 101-privatelink-endpoint-sql/azuredeploy.json:52]: I have asked the product group and this Private DNS Zone name is for public Azure , so Im going to limit this quick start for public cloud

Comment (bmoore-msft) [reply 101-privatelink-endpoint-sql/azuredeploy.json:52]: ok, I'm restoring the badge so we can communicate that it's not supported in FF

**Pull request merged**